controlplane/pkg/model/model.go

```diff
@@ -22,14 +22,28 @@ type Dataplane struct {
 	LocalMechanisms  []*local.Mechanism
 	RemoteMechanisms []*remote.Mechanism
 }
+
+type Endpoint struct {
+	Endpoint       *registry.NSERegistration
+	SocketLocation string
+	Workspace      string
+}
+
 type ClientConnectionState int8
 
 const (
-	ClientConnection_Ready      = 0
-	ClientConnection_Requesting = 1
-	ClientConnection_Healing    = 2
-	ClientConnection_Closing    = 3
-	ClientConnection_Closed     = 4
+	ClientConnection_Ready      ClientConnectionState   = 0
+	ClientConnection_Requesting ClientConnectionState   = 1
+	ClientConnection_Healing    ClientConnectionState   = 2
+	ClientConnection_Closing    ClientConnectionState   = 3
+	ClientConnection_Closed     ClientConnectionState = 4
+)
+
+type DataplaneState int8
+
+const (
+	DataplaneState_None  = 0 // In case dataplane is not yet configured for connection
+	DataplaneState_Ready = 1 // In case dataplane is configured for connection.
```

Comment:
SHould these be of type `DataplaneState`?

Reply:
Yes, Thanks.